# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxMacPlatformFontList.mm
# Commit: 6af07e63a001
# Full Hash: 6af07e63a001dfc667a952727551a80b16797339
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-08-13 09:54:32
# Regressor Bug: 1565966
# File Overlap Count: 1
# Description:
#   Bug 1565966 - part 2 - Include Family attributes in mAliasTable when collecting font-name aliases, so that fontEntry instantiation on Windows can work correctly for the Aliases eventually stored in the font-list. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D41243
# ==============================================================================

diff -r 277876c31869 -r 6af07e63a001 gfx/thebes/gfxMacPlatformFontList.mm
--- a/gfx/thebes/gfxMacPlatformFontList.mm	Mon Aug 12 17:49:09 2019 +0000
+++ b/gfx/thebes/gfxMacPlatformFontList.mm	Mon Aug 12 17:49:28 2019 +0000
@@ -1071,8 +1071,10 @@
           MOZ_ASSERT(false, "single-face family already known");
           break;
         }
-        auto af = mAliasTable.LookupOrAdd(key);
-        af->AppendElement(facePtrs[i]);
+        auto aliasData = mAliasTable.LookupOrAdd(key);
+        // The "alias" here isn't based on an existing family, so we don't call
+        // aliasData->InitFromFamily(); the various flags are left as defaults.
+        aliasData->mFaces.AppendElement(facePtrs[i]);
         break;
       }
     }
@@ -1794,8 +1796,9 @@
     for (const auto& alias : otherFamilyNames) {
       nsAutoCString key;
       GenerateFontListKey(alias, key);
-      auto af = mAliasTable.LookupOrAdd(key);
-      af->AppendElement(facePtrs[i]);
+      auto aliasData = mAliasTable.LookupOrAdd(key);
+      aliasData->InitFromFamily(aFamily);
+      aliasData->mFaces.AppendElement(facePtrs[i]);
     }
   }
 }